synapse/http/proxyagent.py

```diff
@@ -39,6 +39,8 @@ class ProxyAgent(_AgentBase):
         reactor: twisted reactor to place outgoing
             connections.
 
+        proxy_reactor: twisted reactor to use for connections to the proxy server
```

Comment:
Can we add a note about why this might differ from the above? Something like:

> The reactor might block connections to certain IP addresses, but the proxy_reactor should always connect.

Or something like that 🤷